sycl/test-e2e/Matrix/joint_matrix_apply_two_matrices_impl.hpp

```diff
@@ -0,0 +1,147 @@
+//==------- joint_matrix_apply_two_matrices_impl.hpp  - DPC++
+// joint_matrix---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+template <typename Tc, typename Ta, size_t M, size_t N>
+bool assert_ops_ref(Tc *C, Tc *D, Ta *A, Ta *Ar) {
+  for (size_t i = 0; i < M; i++)
+    for (size_t j = 0; j < N; j++) {
+      Tc diffc = D[i * N + j] - C[i * N + j] * 2;
+      Ta diffa = Ar[i * N + j] - (A[i * N + j] + 42);
+      if constexpr (std::is_same_v<Ta, bfloat16>) {
+        if (std::fabs(diffc) > FLOAT_EPSILON ||
+            std::fabs(diffa) > FLOAT_EPSILON) {
```

Comment:
if any operand is nan (it may happen if IGC code did not work correctly), then this check may still return true and test will pass.
```suggestion
        if (std::fabs(diffc) > FLOAT_EPSILON ||
            std::fabs(diffa) > FLOAT_EPSILON || std::isnan(C[i*N+j) || std::isnan(A[i*N+j])) {
```